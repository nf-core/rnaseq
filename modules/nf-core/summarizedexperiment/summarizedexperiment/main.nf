process SUMMARIZEDEXPERIMENT_SUMMARIZEDEXPERIMENT {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bioconductor-summarizedexperiment:1.32.0--r43hdfd78af_0' :
        'biocontainers/bioconductor-summarizedexperiment:1.32.0--r43hdfd78af_0' }"

    input:
    tuple val(meta), path(matrix_files)
    tuple val(meta2), path(rowdata)
    tuple val(meta3), path(coldata)

    output:
    tuple val(meta), path("*.rds")              , emit: rds
    tuple val(meta), path("*.R_sessionInfo.log"), emit: log
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'summarizedexperiment.r'

    stub:
    """
    touch ${meta.id}.SummarizedExperiment.rds
    touch ${meta.id}.R_sessionInfo.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r: \$( R --version | sed '1!d; s/.*version //; s/ .*//' )
    END_VERSIONS
    """
}
