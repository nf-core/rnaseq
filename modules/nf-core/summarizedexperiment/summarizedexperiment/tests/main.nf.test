nextflow_process {

    name "Test Process SUMMARIZEDEXPERIMENT_SUMMARIZEDEXPERIMENT"
    script "../main.nf"
    process "SUMMARIZEDEXPERIMENT_SUMMARIZEDEXPERIMENT"

    tag "modules"
    tag "modules_nfcore"
    tag "custom/tx2gene"
    tag "tximeta"
    tag "tximeta/tximport"
    tag "summarizedexperiment"
    tag "summarizedexperiment/summarizedexperiment"
    tag "untar"

    setup {

        run("UNTAR") {
            script "../../../untar/main.nf"
            process {
                """
                input[0] = [
                [ id:'test'], // meta map
                file(params.test_data['saccharomyces_cerevisiae']['genome']['kallisto_results'], checkIfExists: true)
                ]
                """
            }
        }
        run("CUSTOM_TX2GENE") { 
            script "../../../custom/tx2gene/main.nf"
            process {
                """
                input[0] = [
                [ id:'test'], // meta map
                file(params.test_data['saccharomyces_cerevisiae']['genome']['genome_gfp_gtf'], checkIfExists: true)
                ]
                input[1] = UNTAR.out.untar.map { meta, dir -> [ meta, dir.listFiles().collect() ] }
                input[2] = 'kallisto'
                input[3] = 'gene_id' 
                input[4] = 'gene_name'
                """
            }
        }
        run("TXIMETA_TXIMPORT") {
            script "../../../tximeta/tximport/main.nf"
            process {
                """
                input[0] = UNTAR.out.untar.map { meta, dir -> [ meta, dir.listFiles().collect() ] }
                input[1] = CUSTOM_TX2GENE.out.tx2gene
                input[2] = 'kallisto'
                """
            }
        } 
    }

    test("saccharomyces_cerevisiae - multi_matrix - rowdata - coldata") {
    
        config "./multi_matrix.config"
        
        when {
            process {
                """
                input[0] = TXIMETA_TXIMPORT.out.counts_gene.join(TXIMETA_TXIMPORT.out.tpm_gene).map{tuple(it[0], [it[1], it[2]])}
                input[1] = CUSTOM_TX2GENE.out.tx2gene
		input[2] = [
                    [ id:'gene'], // meta map
                    file(params.test_data['saccharomyces_cerevisiae']['genome']['samplesheet'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            // Some of the log will vary between conda runs- but top bit should be fine
            def loglines = path(process.out.log[0][1]).readLines()[0..14]

            assertAll(
                { assert process.success },
                { assert snapshot(loglines).md5().match("gene_log_multi_matrix_rowdata_coldata") },
                { assert snapshot(process.out.versions).match('versions_multi_matrix_rowdata_coldata') }
            )
        }
    }
    
    test("saccharomyces_cerevisiae - multi_matrix - rowdata") {
        
        config "./multi_matrix.config"
        
        when {
            process {
                """
                input[0] = TXIMETA_TXIMPORT.out.counts_gene.join(TXIMETA_TXIMPORT.out.tpm_gene).map{tuple(it[0], [it[1], it[2]])}
                input[1] = CUSTOM_TX2GENE.out.tx2gene
                input[2] = Channel.of([ [], [] ])
                """
            }
        }

        then {
            // Some of the log will vary between conda runs- but top bit should be fine
            def loglines = path(process.out.log[0][1]).readLines()[0..14]

            assertAll(
                { assert process.success },
                { assert snapshot(loglines).md5().match("gene_log_multi_matrix_rowdata") },
                { assert snapshot(process.out.versions).match('versions_multi_matrix_rowdata') }
            )
        }
    }
    
    test("saccharomyces_cerevisiae - multi_matrix") {
        
        config "./multi_matrix.config"
        
        when {
            process {
                """
                input[0] = TXIMETA_TXIMPORT.out.counts_gene.join(TXIMETA_TXIMPORT.out.tpm_gene).map{tuple(it[0], [it[1], it[2]])}
                input[1] = Channel.of([ [], [] ])
                input[2] = Channel.of([ [], [] ])
                """
            }
        }

        then {
            // Some of the log will vary between conda runs- but top bit should be fine
            def loglines = path(process.out.log[0][1]).readLines()[0..14]

            assertAll(
                { assert process.success },
                { assert snapshot(loglines).md5().match("gene_log_multi_matrix") },
                { assert snapshot(process.out.versions).match('versions_multi_matrix') }
            )
        }
    }
    
    test("saccharomyces_cerevisiae - single_matrix") {
        
        config "./single_matrix.config"
        
        when {
            process {
                """
                input[0] = TXIMETA_TXIMPORT.out.counts_gene
                input[1] = Channel.of([ [], [] ])
                input[2] = Channel.of([ [], [] ])
                """
            }
        }

        then {
            // Some of the log will vary between conda runs- but top bit should be fine
            def loglines = path(process.out.log[0][1]).readLines()[0..14]

            assertAll(
                { assert process.success },
                { assert snapshot(loglines).md5().match("gene_log_single_matrix") },
                { assert snapshot(process.out.versions).match('versions_single_matrix') }
            )
        }
    }

    test("saccharomyces_cerevisiae - multi_matrix - rowdata - coldata - stub") {
    
        config "./multi_matrix.config"
        options "-stub"
        
        when {
            process {
                """
                input[0] = TXIMETA_TXIMPORT.out.counts_gene.join(TXIMETA_TXIMPORT.out.tpm_gene).map{tuple(it[0], [it[1], it[2]])}
                input[1] = CUSTOM_TX2GENE.out.tx2gene
		input[2] = [
                    [ id:'gene'], // meta map
                    file(params.test_data['saccharomyces_cerevisiae']['genome']['samplesheet'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.log).match("gene_log_multi_matrix_rowdata_coldataa_stub") },
                { assert snapshot(process.out.versions).match('versions_multi_matrix_rowdata_coldata_stub') }
            )
        }
    }
    
    test("saccharomyces_cerevisiae - multi_matrix - rowdata - stub") {
        
        config "./multi_matrix.config"
        options "-stub"
        
        when {
            process {
                """
                input[0] = TXIMETA_TXIMPORT.out.counts_gene.join(TXIMETA_TXIMPORT.out.tpm_gene).map{tuple(it[0], [it[1], it[2]])}
                input[1] = CUSTOM_TX2GENE.out.tx2gene
                input[2] = Channel.of([ [], [] ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.log).match("gene_log_multi_matrix_rowdata_stub") },
                { assert snapshot(process.out.versions).match('versions_multi_matrix_rowdata_stub') }
            )
        }
    }
    
    test("saccharomyces_cerevisiae - multi_matrix - stub") {
        
        config "./multi_matrix.config"
        options "-stub"
        
        when {
            process {
                """
                input[0] = TXIMETA_TXIMPORT.out.counts_gene.join(TXIMETA_TXIMPORT.out.tpm_gene).map{tuple(it[0], [it[1], it[2]])}
                input[1] = Channel.of([ [], [] ])
                input[2] = Channel.of([ [], [] ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.log).match("gene_log_multi_matrix_stub") },
                { assert snapshot(process.out.versions).match('versions_multi_matrix_stub') }
            )
        }
    }
    
    test("saccharomyces_cerevisiae - single_matrix - stub") {
        
        config "./single_matrix.config"
        options "-stub"
        
        when {
            process {
                """
                input[0] = TXIMETA_TXIMPORT.out.counts_gene
                input[1] = Channel.of([ [], [] ])
                input[2] = Channel.of([ [], [] ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.log).match("gene_log_single_matrix_stub") },
                { assert snapshot(process.out.versions).match('versions_single_matrix_stub') }
            )
        }
    }
}

