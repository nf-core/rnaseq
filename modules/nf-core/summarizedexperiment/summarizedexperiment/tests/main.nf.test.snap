{
    "gene_log_multi_matrix_rowdata_coldataa_stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T13:57:39.545403663"
    },
    "gene_log_single_matrix_stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T13:58:23.258210775"
    },
    "versions_single_matrix": {
        "content": [
            [
                "versions.yml:md5,1fdcee094bb7ec091b4e445b933d0db3"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T14:03:08.335086603"
    },
    "gene_log_multi_matrix_rowdata_stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T13:57:52.234643863"
    },
    "gene_log_multi_matrix_rowdata": {
        "content": "7c35131fdd46dcdba6c71f7f11d5b2c7",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T14:02:24.430008757"
    },
    "versions_multi_matrix_rowdata_stub": {
        "content": [
            [
                "versions.yml:md5,290f4dc3e123c9468040095de6ffe2e2"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T13:57:52.239867923"
    },
    "versions_multi_matrix_stub": {
        "content": [
            [
                "versions.yml:md5,290f4dc3e123c9468040095de6ffe2e2"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T13:58:07.911717413"
    },
    "versions_multi_matrix_rowdata_coldata_stub": {
        "content": [
            [
                "versions.yml:md5,290f4dc3e123c9468040095de6ffe2e2"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T13:57:39.554437332"
    },
    "gene_log_multi_matrix": {
        "content": "7c35131fdd46dcdba6c71f7f11d5b2c7",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T14:02:46.367962336"
    },
    "gene_log_multi_matrix_stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T13:58:07.904228634"
    },
    "gene_log_single_matrix": {
        "content": "7c35131fdd46dcdba6c71f7f11d5b2c7",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T14:03:08.330328893"
    },
    "versions_single_matrix_stub": {
        "content": [
            [
                "versions.yml:md5,290f4dc3e123c9468040095de6ffe2e2"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T13:58:23.263373495"
    },
    "versions_multi_matrix_rowdata": {
        "content": [
            [
                "versions.yml:md5,1fdcee094bb7ec091b4e445b933d0db3"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T14:02:24.434806727"
    },
    "gene_log_multi_matrix_rowdata_coldata": {
        "content": "7c35131fdd46dcdba6c71f7f11d5b2c7",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T14:02:01.244907114"
    },
    "versions_multi_matrix": {
        "content": [
            [
                "versions.yml:md5,1fdcee094bb7ec091b4e445b933d0db3"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T14:02:46.374127565"
    },
    "versions_multi_matrix_rowdata_coldata": {
        "content": [
            [
                "versions.yml:md5,1fdcee094bb7ec091b4e445b933d0db3"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T14:02:01.257461683"
    }
}